hbase-server/src/main/java/org/apache/hadoop/hbase/fs/ErasureCodingUtils.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.fs;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.stream.Collectors;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hbase.HBaseIOException;
+import org.apache.hadoop.hdfs.DistributedFileSystem;
+import org.apache.hadoop.hdfs.protocol.ErasureCodingPolicyInfo;
+import org.apache.yetus.audience.InterfaceAudience;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@InterfaceAudience.Private
+public final class ErasureCodingUtils {
+
+  private ErasureCodingUtils() {
+  }
+
+  private static final Logger LOG = LoggerFactory.getLogger(ErasureCodingUtils.class);
+
+  /**
+   * Runs checks against the FileSystem, verifying that HDFS is supported and the policy is
+   * available and enabled.
+   */
+  public static void checkAvailable(FileSystem fs, String policy) throws HBaseIOException {
+    DistributedFileSystem dfs = getDfs(fs);
+
+    Collection<ErasureCodingPolicyInfo> policies;
+    try {
+      policies = dfs.getAllErasureCodingPolicies();
```

Comment:
Should we provide a utility or api to get a list of supported and available policies which can be used before some one invokes set ECP? Or is it overkill? Please let me know what do you think.

Reply:
I'd rather not. I thought of this PR being the bare minimum support for safely maintaining the EC state in hbase, there might be other UX improvements we can do over time but out of scope here. Someone using EC should be familiar with the hdfs docs on the matter. We can point to this in the guide (which i'll do in a separate pr)

Reply:
Sounds good